### PR TITLE
Add sendThreads property to `BugsnagConfiguration`

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds the `sendThreads` to `BugsnagConfiguration`. This allows a user to record thread information always, for unhandled errors only, or never.

Note that the stacktrace of the crashing thread is always recorded and reported in the thread tab.

## Changeset

- Added publicly visible `BSGThreadSendPolicy` enum and `sendThreads` property
- Passed value of `sendThreads` into KSCrash during its initialisation
- Altered KSCrash to take an int rather than a bool for the `threadTracingEnabled` value
- Added logic to only serialize all threads if the user has specified to do so

## Tests

- Added unit tests to verify default value and copying of thread information in `BugsnagConfiguration`
- Manually verified behaviour in an example app

### CODE REVIEW

File: Source/BugsnagCrashSentry.m [hunk 1]
```diff
@@ -44,7 +46,7 @@ - (void)install:(BugsnagConfiguration *)config
     
     bsg_kscrash_setHandlingCrashTypes(crashTypes);
     
-    if (![[BSG_KSCrash sharedInstance] install])
+    if (![ksCrash install])
         bsg_log_err(@"Failed to install crash handler. No exceptions will be reported!");
```
Comment (robinmacharg) [inline Source/BugsnagCrashSentry.m:50]: nit: braces round conditional.  Also, while this is being modified, would a rename to e.g. `tryInstall` read better?  I'm slightly swithering on that.

File: Source/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry.h [hunk 2]
```diff
@@ -60,7 +60,7 @@ typedef struct BSG_KSCrash_SentryContext {
     bool reportWhenDebuggerIsAttached;
 
     /** If true, will trace threads and report binary images. */
```
Comment (robinmacharg) [inline Source/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry.h:62]: nit: The comment's not quite accurate w.r.t. the enum

Comment (fractalwrench) [reply Source/KSCrash/Source/KSCrash/Recording/Sentry/BSG_KSCrashSentry.h:62]: Updated comment.

File: Source/BugsnagCrashSentry.m (hunk 1 above)
Comment (fractalwrench) [reply Source/BugsnagCrashSentry.m:50]: Added braces - I'll keep the name as `install` for now as it reduces the size of the changeset, and it's something we can alter later if needed.

**Pull request merged**